### PR TITLE
Upgraded module to hapi v17

### PR DESCRIPTION
**Pull request by mrlannigan**
# Breaking Changes

* Upgraded to hapi 17
* Requires node 8.9 or greater

## Notable Changes

* ~~Replaced Async with Bluebird 3.x.x~~ Removed all production dependencies
* Upgrade modules
  * code 5.x.x
  * lab 15.x.x
* Closes #58
* Closes #57 
* Closes #56

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,41 +1,33 @@
...
+const Promise = require('bluebird');
 const Fs = require('fs');
-const Async = require('async');
 const Path = require('path');
 
 const testDefault = /\.(js)$/;
+const internals = {};
 
-exports.register = (server, baseOptions, next) => {
+internals.readDir = Promise.promisify(Fs.readdir, { context: Fs });
```
Comment (AdriVanHoudt) [inline lib/index.js]: Did you know that node has this built in? See https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original
Please use that one, no need for more deps ^^

File: lib/index.js [hunk 2]
```diff
@@ -1,41 +1,33 @@
...
-    return Fs.readdir(baseOptions.dir, (err, list) => {
+module.exports = {
+    pkg: require('../package.json'),
+    register: async (server, options) => {
 
-        if (err) {
-            return next(err);
-        }
-
-        return Async.each(list, (file, callback) => {
```
Comment (AdriVanHoudt) [inline lib/index.js]: No clue why Async was used lol

File: lib/index.js [hunk 3]
```diff
@@ -1,41 +1,33 @@
...
-            // Skip files with unknown filename extension
-            if (!file.match(baseOptions.test || testDefault)) {
-                return callback();
-            }
+        const fileList = await internals.readDir(options.dir);
+        const fileTest = options.test || testDefault;
 
-            const modulePath = Path.join(baseOptions.dir, file);
-            const routeFile = require(modulePath);
+        for (let i = 0; i < fileList.length; ++i) {
```
Comment (AdriVanHoudt) [inline lib/index.js]: Use `fileList.forEach`

**Comment by mrlannigan** [comment]
@AdriVanHoudt All changes requested have been updated.

Yea I put it in our boilerplate to help folks here have a clear place to look when registering routes.

**Comment by AdriVanHoudt** [comment]
Awesome 🔥

**Comment by AdriVanHoudt** [comment]
Published as 4.0.0 🎉

**Pull request merged**